src/main/java/se/sundsvall/invoices/integration/idata/configuration/AuthorizationInterceptor.java

```diff
@@ -0,0 +1,75 @@
+package se.sundsvall.invoices.integration.idata.configuration;
+
+import feign.RequestInterceptor;
+import feign.RequestTemplate;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Interceptor for adding authorization headers to requests made to the IDATA API. This interceptor generates an HMAC
+ * signature using the given request parameters and secret key.
+ */
+public class AuthorizationInterceptor implements RequestInterceptor {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(AuthorizationInterceptor.class);
+	private static final String AUTHORIZATION_HEADER = "Authorization";
+	private final String apiKey;
+	private final String secretKey;
+
+	public AuthorizationInterceptor(final String apiKey, final String secretKey) {
+		this.apiKey = apiKey;
+		this.secretKey = secretKey;
+	}
+
+	@Override
+	public void apply(RequestTemplate template) {
+		var message = createMessageWithParameters(template.queries());
+
+		var signature = generateHMAC(message, secretKey);
+
+		template.header(AUTHORIZATION_HEADER, "IDATA " + apiKey + ":" + signature);
+	}
+
+	/**
+	 * Helper method that extracts the query parameters from the request template and creates a message string. This string
+	 * is used to generate the HMAC signature.
+	 *
+	 * @param  queries the query parameters from the request template
+	 * @return         the message string
+	 */
+	String createMessageWithParameters(final Map<String, Collection<String>> queries) {
+		return queries.entrySet().stream()
+			.sorted(Map.Entry.comparingByKey())
+			.map(entry -> entry.getKey() + "=" + String.join(",", entry.getValue()))
+			.collect(Collectors.joining("&"));
+	}
+
+	/**
+	 * Generates an HMAC signature using the provided message and key.
+	 *
+	 * @param  message the message to be signed
+	 * @param  key     the secret key used for signing
+	 * @return         the HMAC signature as a hexadecimal string
+	 */
+	String generateHMAC(final String message, final String key) {
+		try {
+			var secretKeySpec = new SecretKeySpec(key.getBytes(), "HmacSHA1");
+			var mac = Mac.getInstance("HmacSHA1");
+			mac.init(secretKeySpec);
+
+			return Hex.encodeHexString(mac.doFinal(message.getBytes()));
+		} catch (NoSuchAlgorithmException | InvalidKeyException e) {
+			LOGGER.error("Error generating HMAC: {}", e.getMessage());
+			return null;
```

Comment:
Should this method return null or throw Problem.valueOf(500)...? In the end the result will be the same I guess, but with different error codes. 

- If null is returned, a 401 or 403 is probably returned from ICASE, resulting in a 502 - Bad Gateway from this service, which is somewhat difficult to interpret into a failure to generate the auth header without some digging
- If we throw an explicit 500 exception with a proper error text (i.e. something like "error occured when trying to generate authorization header") the service will return 500 - Internal Error from this service with an explicit message of what went wrong